EmployeeLib/Manager.cs

```diff
@@ -2,15 +2,17 @@
 
 namespace EmployeeLib
 {
-    public class Manager : Employee
+    public class Manager : EmployeeBase
     {
+        public EmployeeBase Supervisor { get; set; }
```

Comment:
if the manager does have a manager CEO for example, you need to implement ISupervised